### ISSUE TITLE
How to generate bearer once and execute 1000 runs

### ISSUE THREAD

**Issue by nikhilagrawal577** [naver/ngrinder#559]
I am using groovy script for my usecase. I need to generate bearer token from an HTTP API and use it in the script. 

Usecase: For a test where i will hit an API for 10000 times, script should generate the token as first step and then hit that api for 10000 times.

I tried putting the http call to generate token in before process and before thread as well. For each call, its generating new bearer token.

Is there any way that we can configure the script where it will generate the token once use for all runs ??

**Comment by donggyu04** [comment]
@nikhilagrawal577 

`@BeforeProess` is just called one time when before each process is invoked.
so, code in `@BeforeProess` not run every times.

Could you share script that reproduces the problem?

**Comment by imbyungjun** [comment]
You can use `@Before`. The method marked with `@Before` will be invoked once before each test runs.

**Comment by nikhilagrawal577** [comment]
@donggyu04 i am running via a test plan where run count kept at 10000.
Keeping in `beforeprocess`might help for a run count of 1.

**Comment by nikhilagrawal577** [comment]
@imbyungjun Exactly. But if i am running a test for run count of 1000, it calls before process for the same time and it didnt serve the purpose
